### PR TITLE
Filter cln htlcs on intercept

### PR DESCRIPTION
**Pull request by okjodom**
Instead of processing all htlcs to gateway for processing, filter only the relevant ones, and forward all other intercepted htlcs. This rightly assumes the gateway is only serving one federation, and clients in that federation use `short_channel_id=0` in the route hint

**TODO**: Expand this filtering logic to work with other short channel ids, as part of completing #664

### CODE REVIEW

**Comment by dpc** [comment]
Cherry-picked into https://github.com/fedimint/fedimint/pull/1381 to see if helps.

**Comment by okjodom** [comment]
> Cherry-picked into #1381 to see if helps.

@justinmoon, @dpc, you can cherry pick this latest, or review and merge. I figured the correct bolt 4 failure message code to use in case we fail to buy a preimage. Now the node sends this message back to LN in case it needs to gracefully cancel HTLCs

```
2023-01-18T01:40:51.301Z INFO    020901ddfafcad5adc88757f9a09f80f8d1660f29f3461677413df191143fe9623-chan#1: htlc 0 failed from 0th node with code 0x1639 (INVALID 5689)
2023-01-18T01:40:51.314Z UNUSUAL plugin-pay: cmd 234 partid 0: Intermediate node 020901ddfafcad5adc88757f9a09f80f8d1660f29f3461677413df191143fe9623 reported strange error code 1639 (UPDATE|1593)
2023-01-18T01:40:51.314Z INFO    plugin-pay: cmd 234 partid 0: failed: INVALID 5689 (reply from remote)
```

File: gateway/ln-gateway/src/cln.rs [hunk 1]
```diff
@@ -133,22 +132,51 @@ impl LnRpc for Mutex<cln_rpc::ClnRpc> {
...
+/// BOLT 4: https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages
+/// 16399 error code reports unknown payment details.
+///
+/// TODO: We should probably use a more specific error code based on htlc processing fail reason
+fn htlc_processing_failure() -> serde_json::Value {
+    serde_json::json!({
+        "result": "fail",
+        "failure_message": "1639"
+    })
+}
```
Comment (justinmoon) [inline gateway/ln-gateway/src/cln.rs:144]: Can this be a `const`?

**Review by justinmoon — APPROVED** [review]
LGTM

File: gateway/ln-gateway/src/cln.rs (hunk 1 above)
Comment (okjodom) [reply gateway/ln-gateway/src/cln.rs:144]: Tried a const but the strings need to be statics at best. Didn't know how to make it a const

**Pull request merged**